packages/manager/modules/product-offers/src/workflows/product-offers-order-workflow.class.js

```diff
@@ -251,12 +251,21 @@ export default class OrderWorkflow extends Workflow {
     };
 
     if (this.expressOrder) {
-      const productId = this.productName();
+      const productId =
+        typeof this.productName === 'function'
+          ? this.productName()
+          : this.productName;
       const checkoutObject = this.getCheckoutInformations();
+      const { configuration } = checkoutObject;
+      const serviceName =
+        typeof this.serviceNameToAddProduct === 'function'
+          ? this.serviceNameToAddProduct()
+          : this.serviceNameToAddProduct;
       const jsUrlToSend = {
         productId,
-        configuration: checkoutObject.configuration,
+        ...(configuration.length > 0 ? { configuration } : []),
```

Comment:
`configuration.length > 0 ` means you always have an empty array at list (can't be null). Why creating a copy here ?
```suggestion
        ...{ configuration },
```

Reply:
Yes we always have an array (never null)
![image](https://github.com/ovh/manager/assets/8767496/033c7374-9915-41ce-a785-07efeffd3ffe)

Reply:
I do not want to have in my generated JSURL a configuration property visible inside that's why I do this weird object or empty array with ternary condition before